### PR TITLE
Code generation for point to direction block

### PR DESCRIPTION
**Pull request by nkiruka**
This PR is to add logic to point to direction blocks.  

**Implementation**
The point to direction block will allow users to set the absolute rotation of the Artist.  This eliminates the need for a student to calculate how to reset the Artist to a specific direction.  The point to direction does not keep track of the current state of the Artist.

The default rotation is set to clockwise rotation, since the Artist's  initial position is facing east.  The starting value is 0 degrees.

code_generation_pointto

**Observation/Question**
-Two tests were written for pointTo direction blocks.  However, there are no interesting edge cases tested.  Testing setHeading function will probably be a good way to do this.

-Will a writing a test for setHeading be beneficial for this PR?  This will allow for more interesting test cases such as passing in a negative angle, an angle greater than 360, or an angle with invalid characters.

### CODE REVIEW

File: apps/src/turtle/turtle.js [hunk 1]
```diff
@@ -394,6 +396,10 @@ class Visualization {
     }
   }
 
+  pointTo(degrees) {
+    this.setHeading(degrees + ABSOLUTE_DIRECTION);
```
Comment (balderdash) [inline apps/src/turtle/turtle.js]: `DEFAULT_DIRECTION` would make more sense here, and that constant already exists (fortunately with the same value).

File: apps/test/unit/turtle/turtleTest.js [hunk 2]
```diff
@@ -175,6 +175,33 @@ describe('Artist', () => {
...
   });
 
+  describe('pointTo', () => {
+    it('can point to 45 degrees', () => {
+      let angle = 45;
+      let blockId = "block_id_4";
+      let artist = new Artist();
+
+      artist.visualization = new Artist.Visualization();
+      const pointToSpy = sinon.spy(artist.visualization, 'pointTo');
```
Comment (joshlory) [inline apps/test/unit/turtle/turtleTest.js]: Could spy `setHeading` as well for completeness, but this is fine as-is.

File: apps/test/unit/turtle/turtleTest.js [hunk 3]
```diff
@@ -175,6 +175,33 @@ describe('Artist', () => {
...
+      expect(pointToSpy).to.be.have.been.calledWith(angle);
+    });
+
+    it('can point to a specific direction', () => {
+      const angle = 45;
+      const blockId = "block_id_8";
+      let artist = new Artist();
+
+      artist.visualization = new Artist.Visualization();
+      console.info(typeof artist.visualization.pointTo);
```
Comment (joshlory) [inline apps/test/unit/turtle/turtleTest.js]: Extra?

File: apps/src/turtle/turtle.js [hunk 4]
```diff
@@ -1527,6 +1533,9 @@ Artist.prototype.step = function (command, values, options) {
       tupleDone = result.tupleDone;
       this.visualization.turnByDegrees(result.distance);
       break;
+    case 'PT': //Point To
```
Comment (joshlory) [inline apps/src/turtle/turtle.js]: Tiny style nit: space after `//`

File: apps/test/unit/turtle/turtleTest.js [hunk 5]
```diff
@@ -175,6 +175,32 @@ describe('Artist', () => {
...
+      expect(pointToSpy).to.be.have.been.calledWith(angle);
+    });
+
+    it('can point to a specific direction', () => {
+      const angle = 45;
+      const blockId = "block_id_8";
+      let artist = new Artist();
+
+      artist.visualization = new Artist.Visualization();
+      artist.visualization.angle = 45;
```
Comment (balderdash) [inline apps/test/unit/turtle/turtleTest.js]: This should be something other than 45, to ensure that the `PT` step actually changes the angle.

File: apps/test/unit/turtle/turtleTest.js (hunk 3 above)
Comment (nkiruka) [reply apps/test/unit/turtle/turtleTest.js]: Fixed.  Thanks

File: apps/src/turtle/turtle.js (hunk 1 above)
Comment (nkiruka) [reply apps/src/turtle/turtle.js]: Fixed.  Thanks

File: apps/src/turtle/turtle.js (hunk 4 above)
Comment (nkiruka) [reply apps/src/turtle/turtle.js]: Fixed.  Thanks

File: apps/test/unit/turtle/turtleTest.js (hunk 2 above)
Comment (nkiruka) [reply apps/test/unit/turtle/turtleTest.js]: Test added to stub setHeading.

File: apps/test/unit/turtle/turtleTest.js (hunk 5 above)
Comment (nkiruka) [reply apps/test/unit/turtle/turtleTest.js]: Modified test to include different angles.

**Pull request merged**